front_end/src/actions/Accounts/AccountActions.js

```diff
@@ -40,7 +47,71 @@ export function requestAccount() {
 export function receiveAccount(json) {
   return {
     type: RECEIVE_ACCOUNT,
-    details: json.result
+    details: json
+  };
+}
+
+export function requestAccounts() {
+  return {type: REQUEST_ACCOUNTS};
+}
+export function receiveAccounts(json) {
+  return {
+    type: RECEIVE_ACCOUNTS,
+    rows: json
+  };
+}
+
+export function createAccount(data) {
+  // thunk middleware knows how to handle functions
+  return function next(dispatch) {
+    dispatch(requestCreateAccount());
+    // Return a promise to wait for
+    return fetch(apiUrl + '/api/accounts', {
+      method: 'post',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: data
+    })
+      .then(response => response.json())
+      .then((json) => new Promise(resolve => {
+        if(json.result !== undefined){
+          dispatch(receiveCreateAccount(json.result));
+        }
+        else{
+          dispatch(receiveCreateAccount(null));
+        }
+        resolve(json);
+      })
+    );
+  };
+}
+
+export function updateAccount(accountId, data) {
+  // thunk middleware knows how to handle functions
+  return function next(dispatch) {
+    dispatch(requestUpdateAccount());
+    // Return a promise to wait for
+    return fetch(apiUrl + '/api/accounts/' + accountId, {
+      method: 'put',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: data
+    })
+      .then(response => response.json())
+      .then((json) => new Promise(resolve => {
```

Comment:
Super nitpick because I haven't found anything to pick on yet. There parens around `json` aren't needed and you didn't have them above around `response` so it bothers my eyes a little. Consistency FTW